tutormfe/templates/mfe/build/mfe/Dockerfile

```diff
@@ -56,18 +56,40 @@ RUN /openedx/i18n/i18n-merge.js /openedx/app/src/i18n/messages /openedx/i18n/{{
 FROM base AS {{ app_name }}-common
 COPY --from={{ app_name }}-src /package.json /openedx/app/package.json
 COPY --from={{ app_name }}-src /package-lock.json /openedx/app/package-lock.json
+RUN echo "copying files"\
+{%- for domain, theme in iter_domains(GROVE_ADDITIONAL_DOMAINS, MFE_HOST, MFE_BRAND_PACKAGE_NAME) %}
+    && mkdir -p /openedx/app/{{ domain }} \
+    && cp /openedx/app/package.json /openedx/app/{{ domain }}/package.json \
+    && cp /openedx/app/package-lock.json /openedx/app/{{ domain }}/package-lock.json \
+{%- endfor %}
+    && echo "done"
 ARG NPM_REGISTRY={{ NPM_REGISTRY }}
 {{ patch("mfe-dockerfile-pre-npm-install") }}
 {{ patch("mfe-dockerfile-pre-npm-install-{}".format(app_name)) }}
 {#- Required for building optipng on M1 #}
 ENV CPPFLAGS=-DPNG_ARM_NEON_OPT=0
 {#- We define this environment variable to bypass an issue with the installation of pact https://github.com/pact-foundation/pact-js-core/issues/264 #}
 ENV PACT_SKIP_BINARY_INSTALL=true
-RUN {% if is_buildkit_enabled() %}--mount=type=cache,target=/root/.npm,sharing=shared {% endif %}npm clean-install --no-audit --no-fund --registry=$NPM_REGISTRY
+RUN mkdir -p /openedx/themes
+COPY ./themes /openedx/themes
+
+RUN {% if is_buildkit_enabled() %}--mount=type=cache,target=/root/.npm,sharing=shared {% endif %} echo "building MFEs" \
+{%- for domain, theme in iter_domains(GROVE_ADDITIONAL_DOMAINS, MFE_HOST, MFE_BRAND_PACKAGE_NAME) %}
+  && npm clean-install --prefix={{domain}} --no-audit --no-fund --registry=$NPM_REGISTRY \
```

Comment:
@xitij2000 Do we need to do it for every domain if the `package.json` and `package-lock.json` are the same for all of them since they are copied from the base [here](https://github.com/open-craft/tutor-mfe/pull/4/files#diff-d4211adff501de27b7b5207952e99333f754dcd85227d6b1d19bab9e98a1defeR60-R64)? I was wondering if we can also avoid building separately for all domains, i.e. install once, build, copy for each domain, then install the themes and build again. I don't think it would improve docker's caching, but we could avoid rebuilding the same stuff multiple times.

Reply:
Ah, I see that it can be troublesome, since we are only building for the prod images, so my suggested sequence of steps might not work.

Reply:
I think building, installing the theme and then building again won't work currently since webpack will just build everything again even though only the theme has changed. However, this did give me an idea, which is to just install all the dependencies once, copy over to each site folder, then install the theme, then build separately. This will avoid running npm install multiple times when the package contents are the same. It's worked in testing, so I'll push it in a separae commit.